main.py

```diff
@@ -107,7 +107,7 @@ def get_inps(
             (min(nsamples_per_device, len(data) - i * nsamples_per_device), model_seqlen, model.config.hidden_size),
             dtype=dtype,
             device=devices[i] if not offload_activations else "cpu",
-            pin_memory=offload_activations,
+            pin_memory=offload_activations and devices != [torch.device('cpu')],
```

Comment:
Since training works without --offload_activations flag, we don't need this change anymore

Reply:
Without this additional condition we will cause the exception if we run the program having only CPU and giving --offload_activations flag
But anyway, if you think this is better I can change it

Reply:
Thanks. My point is that offload_activations on cpu does not make sense, and shouldn't be supported